### ISSUE TITLE
document a HTTP header param which is only used by interceptor/filter and does not appear in the method signature

### ISSUE THREAD

**Issue by zappee** [swagger-api/swagger-core#2836]
This is a new feature request.
See the details bellow.

I have a Jersey REST method and have a custom filter which is appiled as an annotation on REST methods:

```
@Priority(FilterPriorities.USER_ID_VALIDATOR)
@Provider
@UserIdCheck
public class UserIdValidatorFilter implements ContainerRequestFilter {

    @Override
    public void filter(final ContainerRequestContext requestContext) throws IOException {
        // checks presence of user-id
        ...

        // checks presence of sso-token
        String ssoToken = requestContext.getHeaderString(CustomHttpHeader.SSO_TOKEN);
        ...

        // validates SSO token against used ID
        ...
    }
}
```

This is my REST method:

```
@POST
@Path("/upload")
@Consumes(ExtendedMediaType.MULTIPART_FORM_DATA)
@ApiOperation(
        value = "Saves the uploaded image",
        notes = "Returns the unique id of the image.",
        response = String.class)
@ApiResponses(value = {
        @ApiResponse(code = 200, message = "The uploaded image has been saved.", response = String.class),
        @ApiResponse(code = FileTooLargeException.HTTP_STATUS, message = FileTooLargeException.SWAGGER_API_RESPONSE_MESSAGE, response = ErrorInfo.class),
        @ApiResponse(code = InvalidMediaTypeError.HTTP_STATUS, message = InvalidMediaTypeError.SWAGGER_API_RESPONSE_MESSAGE, response = ErrorInfo.class),
        @ApiResponse(code = MissingHttpHeaderException.HTTP_STATUS, message = MissingHttpHeaderException.SWAGGER_API_RESPONSE_MESSAGE, response = ErrorInfo.class),
        @ApiResponse(code = UnauthorizedRequestException.HTTP_STATUS, message = UnauthorizedRequestException.SWAGGER_API_RESPONSE_MESSAGE, response = ErrorInfo.class)})
@UserIdCheck
@Override
public String uploadImage(
        @ApiParam(value = "id of the authenticated user", required = true) @HeaderParam(CustomHttpHeader.USER_ID) final Long userId,
        @ApiParam(value = "file to upload", required = true) @FormDataParam("file") final InputStream inputStream,
        @ApiParam(value = "details of the uploaded file", required = true) @FormDataParam("file") final FormDataContentDisposition fileDetail) {

        // do something here...
    }
}
```

My custom annotation (`@UserIdCheck`) checks whether the sso token is presented on the HTTP header and it is valid or not. SSO token is not used by the REST method so it does not appear on the method signature BUT it needs to be on the HTTP header otherwise the HTTP request is blocked by the filter/interceptor.

I am not able to make a valid test request via swagger UI because the SSO_TOKEN field does not appear on the Swagger test page. And of course, this important field is missing from the Swagger documentation as well.

I do not want to add this variable to the method signature because it is only used by filter, and I do not need this value in the method body.

Is there any solution to handle this use case?
If not then I would like to ask the comunity to add this feature to the Swagger in the near furure.

**Comment by Marek00Malik** [comment]
This would be a good feature, I have a very similar case in my project. I either need to add the information about this special implicit header in each new endpoint or (and that would be done on the filter) add this information on the filter description.

**Comment by MichakrawSB** [comment]
Closing this issue as part of our https://github.com/swagger-api/swagger-core/issues/5074.
This issue is outdated and/or inactive for a long time, and we’re not able to revisit it meaningfully.

If this problem is still relevant, please open a new issue and reference this one for context.
Thank you for your understanding!
